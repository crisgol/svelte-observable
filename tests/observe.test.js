import test from 'ava';
import Observable from 'zen-observable';
import { load, check } from './helpers';
import { observe } from '../';

test('should have initial pending value', async t => {
  const observable = Observable.of(1, 2, 3);
  const store = observe(observable);

  const values = await load(store);
  const states = await check(values);

  t.true(states[0].pending);
});

test('should have fulfilled values', async t => {
  const observable = Observable.of(1, 2, 3);
  const store = observe(observable);

  const values = await load(store);
  const states = await check(values);

  t.snapshot(states);
});

test('should have rejected error values', async t => {
  const observable = new Observable(observer => {
    observer.error(new Error('Uh oh.'));
  });
  const store = observe(observable);

  const values = await load(store);
  const states = await check(values);

  t.true(states[1].rejected);
  t.is(states[1].error.message, 'Uh oh.');
});

test('should have readable passthrough if not Observable', async t => {
  const store = observe(4);

  const values = await load(store);
  t.is(values[0], 4);
});
