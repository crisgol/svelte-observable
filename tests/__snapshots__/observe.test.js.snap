// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should have fulfilled values 1`] = `
Array [
  Object {
    "fulfilled": true,
    "value": 1,
  },
  Object {
    "fulfilled": true,
    "value": 2,
  },
  Object {
    "fulfilled": true,
    "value": 3,
  },
]
`;
